lib/modules/manager/helmfile/artifacts.spec.ts

```diff
@@ -152,6 +152,157 @@ describe('modules/manager/helmfile/artifacts', () => {
+    fs.readLocalFile.mockResolvedValueOnce(
+      lockFileTwoWithoutRepositories as never
+    );
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache'
+    );
+    fs.getParentDir.mockReturnValue('');
+    const updatedDeps = [{ depName: 'dep1' }, { depName: 'dep2' }];
+    expect(
+      await helmfile.updateArtifacts({
+        packageFileName: 'helmfile.yaml',
+        updatedDeps,
+        newPackageFileContent: helmfileYamlWithoutRepositories,
+        config,
+      })
+    ).toEqual([
+      {
+        file: {
+          type: 'addition',
+          path: 'helmfile.lock',
+          contents: lockFileTwoWithoutRepositories,
+        },
+      },
+    ]);
+    expect(execSnapshots).toMatchObject([
+      { cmd: 'helmfile deps -f helmfile.yaml' },
+    ]);
+  });
+
+  it('log into private OCI registries, returns updated helmfile.lock', async () => {
+    const helmfileYamlOCIPrivateRepo = codeBlock`
+    repositories:
+      - name: private-charts
+        url: ghcr.io/charts
+        oci: true
+    releases:
+      - name: chart
+        chart: private-charts/chart
+        version: 0.12.0
+    `;
+    const lockFileOCIPrivateRepo = codeBlock`
+    version: 0.151.0
+    dependencies:
+    - name: chart
+      repository: oci://ghcr.io/private-charts
+      version: 0.11.0
+    digest: sha256:e284706b71f37b757a536703da4cb148d67901afbf1ab431f7d60a9852ca6eef
+    generated: "2023-03-08T21:32:06.122276997+01:00"
+    `;
+    const lockFileOCIPrivateRepoTwo = codeBlock`
+    version: 0.151.0
+    dependencies:
+    - name: chart
+      repository: oci://ghcr.io/private-charts
+      version: 0.12.0
+    digest: sha256:9d83889176d005effb86041d30c20361625561cbfb439cbd16d7243225bac17c
+    generated: "2023-03-08T21:30:48.273709455+01:00"
+    `;
+    hostRules.add({
+      username: 'test',
+      password: 'password',
+      hostType: 'docker',
+      matchHost: 'ghcr.io',
+    });
+
+    git.getFile.mockResolvedValueOnce(lockFileOCIPrivateRepo as never);
+    fs.getSiblingFileName.mockReturnValueOnce('helmfile.lock');
+    const execSnapshots = mockExecAll();
+    fs.readLocalFile.mockResolvedValueOnce(lockFileOCIPrivateRepoTwo as never);
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache'
+    );
+    fs.getParentDir.mockReturnValue('');
+    const updatedDeps = [{ depName: 'dep1' }, { depName: 'dep2' }];
+    expect(
+      await helmfile.updateArtifacts({
+        packageFileName: 'helmfile.yaml',
+        updatedDeps,
+        newPackageFileContent: helmfileYamlOCIPrivateRepo,
+        config,
+      })
+    ).toEqual([
+      {
+        file: {
+          type: 'addition',
+          path: 'helmfile.lock',
+          contents: lockFileOCIPrivateRepoTwo,
+        },
+      },
+    ]);
+    expect(execSnapshots).toMatchObject([
+      {
+        cmd: 'helm registry login --username test --password password https://ghcr.io',
```

Comment:
```suggestion
        cmd: 'helm registry login --username test --password password ghcr.io',
```
The documentation mentions only `host` and has only examples without the protocol: 
https://helm.sh/docs/topics/registries/#commands-for-working-with-registries

Reply:
Yes, it is indeed not in the example, but I have confirmed that helm registry login succeeds in a local environment with protocol.
There is also a GCP example with protocol.
```
gcloud auth application-default print-access-token | helm registry login -u oauth2accesstoken \
--password-stdin https://LOCATION-docker.pkg.dev
```
https://cloud.google.com/artifact-registry/docs/helm/authentication#token

Reply:
I would prefer to use host and let helm decide which protocol to use. Most of the time it will be irrelevant, but in edge cases ( e.g local setups without TLS ) this would not work.

@rarkins @viceice WDYT?

Reply:
should i fix it?

Reply:
yes, use without protocol